full-node/db/sov-schema-db/src/lib.rs

```diff
@@ -292,58 +62,101 @@ impl SchemaBatch {
         key: &impl KeyCodec<S>,
         value: &impl ValueCodec<S>,
     ) -> anyhow::Result<()> {
-        let _timer = SCHEMADB_BATCH_PUT_LATENCY_SECONDS
+        #[cfg(feature = "std")]
+        let _timer = metrics::SCHEMADB_BATCH_PUT_LATENCY_SECONDS
             .with_label_values(&["unknown"])
             .start_timer();
+
         let key = key.encode_key()?;
         let value = value.encode_value()?;
+
+        #[cfg(feature = "std")]
```

Comment:
We have `cfg_attr(not(feature = "std"), allow(dead_code))]` in line 39.  If the code is not used in `std` why do we have the flag here?

Reply:
I have just noticed that the `row` in `SchemaBatch` is always `Mutex`  from std. We should feature gate the row in `SchemaBatch` instead in the caller.

Reply:
We do have a `Mutex` available for no-std; `SchemaBatch` type definition is available for such environment.

Regarding the `dead_code`, we make the type definition available, but the concrete implementation is done only in `std`, where we interact with a rocksdb backend. I think it's best to make types available for `no-std` so we can represent them, even without certain functionality. Wdyt?